things/mocks/things.go

```diff
@@ -115,7 +115,7 @@ func (trm *thingRepositoryMock) RetrieveAll(_ context.Context, owner string, pm
 	trm.mu.Lock()
 	defer trm.mu.Unlock()
 
-	if pm.Limit <= 0 {
+	if pm.Limit < 0 {
```

Comment:
Any reason to do this?

Reply:
Reason is to be consistent with all other endpoints. Limit and other page parameters (offset, order, dir) are everywhere optional parameters, and if we want to keep them optional here we need default value if they are missing. When missing, limit is zero (`omitempty` option) so we need to keep not throwing error in that case, that is why we change validation to allow it zero and use default value in that case. Same like you omit `limit` in all other endpoints, it's not `0` if you omit it, it's default `10`